src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/MemoryAddressImpl.java

```diff
@@ -35,62 +35,52 @@
  * This class provides an immutable implementation for the {@code MemoryAddress} interface. This class contains information
  * about the segment this address is associated with, as well as an offset into such segment.
  */
-public final class MemoryAddressImpl implements MemoryAddress {
+public abstract class MemoryAddressImpl implements MemoryAddress {
```

Comment:
Add a comment we could mark this class as a sealed class? Dunno if it coveys any performance advantage but it does covey that the class cannot be extended outside of its nest.

Reply:
true